### ISSUE TITLE
Finer object composition for the controller class

### ISSUE THREAD

**Issue by wlandau** [wlandau/crew#84]
In `crew`, every `R6` [controller](https://github.com/wlandau/crew/blob/main/R/crew_controller.R) object contains 2 smaller `R6` objects:

1. A [router](https://github.com/wlandau/crew/blob/main/R/crew_router.R) to keep track of `mirai::daemons()` and the `mirai`  dispatcher, and
2. A [launcher](https://github.com/wlandau/crew/blob/main/R/crew_launcher.R) that knows how to create, monitor, and terminate `mirai` servers on the user's platform of choice. A [launcher plugin](https://wlandau.github.io/crew/articles/launcher_plugins.html) is an `R6` subclass that inherits from the [abstract launcher class](https://github.com/wlandau/crew/blob/main/R/crew_launcher.R).

That leaves a lot for the controller to do: namely,

A. Manage `mirai` task objects (e.g. [`controller$collect()`](https://github.com/wlandau/crew/blob/34800d9f7e47184726b9d48b3399704d8f80fbef/R/crew_controller.R#L397-L422)), and
B. Auto-scale `mirai` servers ([`controller$scale()](https://github.com/wlandau/crew/blob/34800d9f7e47184726b9d48b3399704d8f80fbef/R/crew_controller.R#L240-L266)).

Both A and B are highly nontrivial, and if @shikokuchuo's promising idea from https://github.com/shikokuchuo/mirai/discussions/60#discussioncomment-5936611 comes to fruition, then each will run on different processes. So from `crew`'s side, it makes sense to create `R6` classes for each: a "schedule" for A and a "regulator" for B. The regulator will contain the launcher, and it will be shipped to the dispatcher process in https://github.com/shikokuchuo/mirai/discussions/60#discussioncomment-5936611 to do auto-scaling given a very simple set of inputs.

Currently, the object composition structure looks like this:

```
controller
├── launcher
└── router
```

My goal for the structure is:

```
controller
├── schedule
├── regulator
│   └── launcher
└── router
```

**Comment by wlandau** [comment]
One nice simplification is to have the launcher call `mirai::saisei()`. If auto-scaling becomes possible on the `mirai` dispatcher, a dispatcher-side version of `saisei()` will be needed, as the current `saisei()` seems to be designed for the client only.

**Comment by wlandau** [comment]
Launcher-side `saisei()` and `poll()` work great and simplify things. In fact, I don't think `crew` even needs a "regulator" class. The launcher itself can auto-scale if I calculate demand differently. 

Also, I am thinking of a couple more changes (unfortunately breaking ones for launcher plugins, but `crew` is still new enough for that):

1. Rename "router" to "client"
2. Condense the `launcher`, `worker`, and `instance` arguments of `launch_worker()` down to a single `name` argument with the worker name already fully formed.

**Comment by wlandau** [comment]
Also should bring back `seconds_interval` to launchers to prepare for integration with the `mirai` dispatcher.

**Comment by wlandau** [comment]
In [branch `84`](https://github.com/wlandau/crew/tree/84), I simplified the internals a whole lot. Whether or not `crew` will be able to integrate with the `mirai` dispatcher (which would require a little more work on my end, but nothing painful), I am happy with the heavy refactor.

Now I need to debug and test:

* [ ] Rd examples
* [ ] README examples
* [ ] Vignette examples
* [ ] Launcher tests
* [ ] interactive tests
* [ ] Unit tests
* [ ] throughput tests
* [ ] `crew.cluster` integration
* [ ] `targets` integration

**Comment by wlandau** [comment]
> Condense the launcher, worker, and instance arguments of launch_worker() down to a single name argument with the worker name already fully formed.

No, this didn't work. `crew.cluster` relies on having access to each of these pieces separately. Users can construct job names on their own.

**Comment by wlandau** [comment]
Best of both worlds: supply both the job name and the pieces.

**Comment by wlandau** [comment]
#87 caused #88 and #89, but it at least fixed #84.
